### PR TITLE
Fix metrics reporting

### PR DESCRIPTION
**Pull request by eamsen**
Also reduced nesting and indentation to help with the readability and changed some names and comments for consistency.

### CODE REVIEW

File: lib/plugins/index.js [hunk 1]
```diff
@@ -54,6 +67,23 @@ exports.handleResponse = function(request, source, dest, options) {
...
+    });
+
+    currentDest.on('close', function() {
+      if (!finished) {
+        request.warn('unexpected close in ' + plugin.name);
+        finalize();
+      }
+    });
+
+    currentDest.on('error', function(e) {
```
Comment (snorp) [inline lib/plugins/index.js]: maybe set finished = true here too?

**Comment by snorp** [comment]
This looks good, not sure if the nit is significant or not

File: lib/plugins/index.js (hunk 1 above)
Comment (eamsen) [inline lib/plugins/index.js]: I think I would prefer not to, and keep finished exclusive to the finish event, it doesn't really change any other behavior either way as that's the only place we check for it.

File: lib/plugins/index.js [hunk 2]
```diff
@@ -54,6 +66,23 @@ exports.handleResponse = function(request, source, dest, options) {
 
     shortcut = shortcut || plugin.isShortcut;
 
+    currentDest.on('finish', function() {
+      finished = true;
+      finalize();
+    });
+
+    currentDest.on('close', function() {
+      if (!finished) {
```
Comment (snorp) [inline lib/plugins/index.js]: I've been looking at the stream docs a lot today and you do indeed only get 'close' if it occurs before end (and therefore 'finish'), so you don't need the 'finished = true' stuff.

**Comment by snorp** [comment]
Looks pretty good, but I think you can nuke the 'finished = true' stuff.

File: lib/plugins/index.js (hunk 2 above)
Comment (eamsen) [inline lib/plugins/index.js]: Removing the same type of checks in `lib/proxy.js`, we get `unexpected close` warnings regularly when browsing, the checks reduce them to a few cases, mostly connected to actual error-state connections.
We could remove the checks and investigate the warnings more to get to the ground of this?

**Pull request merged**